bitnami/nginx/templates/deployment.yaml

```diff
@@ -173,6 +173,9 @@ spec:
             - name: https
               containerPort: {{ .Values.containerPorts.https }}
             {{- end }}
+          {{- if .Values.lifecycleHooks }}
+          lifecycle: {{- toYaml .Values.lifecycleHooks | nindent 12 }}
```

Comment:
I saw you mentioned using `toYaml` here and its possible drawbacks. Could you please check if using our default `common.tplvalues.render` function will work better here?

```
{{- define "common.tplvalues.render" -}}
    {{- if typeIs "string" .value }}
        {{- tpl .value .context }}
    {{- else }}
        {{- tpl (.value | toYaml) .context }}
    {{- end }}
{{- end -}}
```

Referring again to our scaffolding templates, we normally use:

```suggestion
          lifecycle: {{- include "common.tplvalues.render" (dict "value" .Values.lifecycleHooks "context" $) | nindent 12 }}
```

Reply:
yes it does work with the default as well

![grafik](https://user-images.githubusercontent.com/94913584/163167044-cd4d35b3-4397-4161-b381-baf2e3266545.png)